src/ferc_xbrl_extractor/instance.py

```diff
@@ -321,13 +323,11 @@ def parse(self, fact_prefix: str = "ferc") -> Instance:
         parser = etree.XMLParser(huge_tree=True)
 
         # Check if instance contains path to file or file data and parse accordingly
-        if isinstance(self.file, str):
+        try:
             tree = etree.parse(self.file, parser=parser)  # nosec: B320
-            root = tree.getroot()
-        elif isinstance(self.file, io.BytesIO):
-            root = etree.fromstring(self.file.read(), parser=parser)  # nosec: B320
-        else:
-            raise TypeError("Can only parse XBRL from path to file or file like object")
+        except etree.XMLSyntaxError:
+            return None
```

Comment:
We could move this try/except into `xbrl.extract`... then this function will just raise an exception, which I think is maybe nicer? Future callers get to interpret the exception instead of us swallowing it here.

Reply:
Tried pulling this out, and found the original place where this handling was occuring in `xbrl.py` so I think it's the right move.